js/src/mpl_widget.js

```diff
@@ -186,7 +188,11 @@ export class MPLCanvasModel extends widgets.DOMWidgetModel {
         this.image.src = image_url;
 
         // Tell Jupyter that the notebook contents must change.
-        this.send_message('ack');
+        if (!this.acknowledged_rendered) {
+            this.send_message('ack');
```

Comment:
Why do we need a round-trip to the front-end??

Reply:
I tried implementing it without the round-trip but I wasn't able. I need to look deeper into that.